pom.xml

```diff
@@ -218,6 +218,12 @@
 			<version>4.12</version>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>org.hamcrest</groupId>
+			<artifactId>hamcrest-all</artifactId>
```

Comment:
Sure.
I've dropped the hamcrest-all dependency and used vanilla junit 4 assertions.

BTW @artembilan  would it make sense to have a conversion of the tests of the project to junit 5 ? I'd be willing to do this.

Reply:
This project is still aimed to support Java 6 and that's why it's Spring base like is 4.x, too.

I'm not sure that it is going to be so easy to move to JUnit 5 and still support Spring 4 and Java 6...

Reply:
According to  https://junit.org/junit5/docs/current/user-guide/#overview-java-versions it needs at least java 8.
In the README.md of the project is not mentioned though a reference about the supported java/spring versions. I guess it would make sense to have this.

Reply:
By the way, in the nearby issue #154 we with @dsyer are experimenting around async retry and CompletableFutures, so it would be generally good to know whether the switching to java8 is impossible due to major backward compatibility issues (and we should consider a separate module), or possible.

Reply:
There is a `ListenableFuture` abstraction in Spring for that reason.
Although it is fully different story and not related to what you are doing here...

Reply:
@ashamukov thanks for pointing out the existence of  async-retry project. I didn't know about it before.